### DISCUSSION TITLE
zigbee2mqtt-fails-to-start

### DISCUSSION THREAD

**Discussion by BobbySoc** [Koenkk/zigbee2mqtt#15540] [Devices]
``As another user affected by this:

A bit of background HassOS on a POi 4 for a conbee 2 usb Stick. I've followed the suggestions in https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html  to no success.

Any ideas? - tried everything I can think of today so now out of ideas.

Changed device names between [/dev](device: /dev/ttyACM0) and the longer device ID one. (/dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2498XXX-if00)

Interestingly the Phoscon / deconz plugin works and shows me values for the one sensor correctly... - but they then don't show up in Home assistant.

Config file for Zigbee2MQTT:

`data_path: /config/zigbee2mqtt
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: false
mqtt: {}
serial:
  device: /dev/ttyACM0
  adapter: deconz
zigbee_herdsman_debug: true
`

Log files:

`[22:20:52] INFO: Preparing to start...
[22:20:52] INFO: Socat not enabled
[22:20:52] INFO: Zigbee Herdsman debug logging enabled
[22:20:54] INFO: Starting Zigbee2MQTT...
Zigbee2MQTT:debug 2022-12-13 22:20:59: Can't load state from file /config/zigbee2mqtt/state.json (doesn't exist)
Zigbee2MQTT:info  2022-12-13 22:20:59: Logging to console and directory: '/config/zigbee2mqtt/log/2022-12-13.22-20-59' filename: log.txt
Zigbee2MQTT:debug 2022-12-13 22:20:59: Removing old log directory '/config/zigbee2mqtt/log/2022-12-13.21-33-53'
Zigbee2MQTT:info  2022-12-13 22:20:59: Starting Zigbee2MQTT version 1.28.4 (commit #unknown)
Zigbee2MQTT:info  2022-12-13 22:20:59: Starting zigbee-herdsman (0.14.76)
Zigbee2MQTT:debug 2022-12-13 22:20:59: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/config/zigbee2mqtt/coordinator_backup.json","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","databasePath":"/config/zigbee2mqtt/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"adapter":"deconz"}}'
2022-12-13T22:21:00.186Z zigbee-herdsman:adapter No path provided, auto detecting path
Zigbee2MQTT:error 2022-12-13 22:21:00: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-12-13 22:21:00: Failed to start zigbee
Zigbee2MQTT:error 2022-12-13 22:21:00: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-12-13 22:21:00: Exiting...
Zigbee2MQTT:error 2022-12-13 22:21:00: Error: spawn udevadm ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)`

Not 100% sure where to go next and more N00b than not, any ideas?

**Comment by KalleKal** [comment]
Dit you put the “adapter: deconz”-part in the config?

Should read:
serial:
port: /dev/ttyACM0 (or wherever you find it)
adapter: deconz

this worked for me.

https://community.home-assistant.io/t/conbee-ii-and-zigbee2mqtt-no-worky/376174/4

**Reply by BobbySoc** [reply]
Sorry for the late reply (been out of town) - will try in the morning…. 

Hopefully swapping port: and device: will work!

thanks- will let you know how I get on!

**Reply by BobbySoc** [reply]
All now works! - thanks @KalleKal (much better when i get port & device typo sorted.

For those affected - with Conbee 2 with Zigbee2MQTT should read (bold for emphasis!)

***********************
serial:
     **port:** / dev/ttyACM0 (or whatever!)
     **adapter:** deconz

***********************

Thanks once again.
